app/controllers/api/base_controller.rb

```diff
@@ -90,5 +91,17 @@ def render_error(status, type, data = {})
         }.merge(data)
       }, status: status
     end
+
+    def log_usage!
+      return unless user_signed_in?
+
+      Thread.new do
+        log_time = Time.current.utc.strftime('%Y-%m-%d %H:%M')
+        path_hash = Digest::SHA1.hexdigest(request.path)
+        redis = Redis.new(url: "#{Exercism.config.tooling_redis_url}/5")
+        redis.incr("#{Exercism.env}:request:#{path_hash}:#{request.method}:#{current_user.id}:#{log_time}")
+        redis.set("#{Exercism.env}:url:#{path_hash}", request.path)
+      end
+    end
```

Comment:
Can you throw a `rescue` before the final end of this method, just in case there's anything that blows up pls (I can't see why there would be, but I always like to do this for temporary code)